plugins/vi-mode/vi-mode.plugin.zsh

```diff
@@ -1,12 +1,29 @@
 VI_KEYMAP=main
 
+function _vi-mode-set-cursor-shape-for-keymap() {
+  # https://vt100.net/docs/vt510-rm/DECSCUSR
+  local _shape=0
+  case "${1:-${VI_KEYMAP:-main}}" in
+    main)    _shape=6 ;; # vi insert: line
+    viins)   _shape=6 ;; # vi insert: line
+    isearch) _shape=6 ;; # inc search: line
+    command) _shape=6 ;; # read a command name
+    vicmd)   _shape=2 ;; # vi cmd: block
+    visual)  _shape=2 ;; # vi visual mode: block
+    viopp)   _shape=0 ;; # vi operation pending: blinking block
```

Comment:
I can't see how this can be working for local keymaps. You're setting `VI_KEYMAP` from `$KEYMAP` which contains the global keymap, and not the local keymap. `zle-keymap-select` will never be called with `$KEYMAP` set to a local keymap like `viopp` or `visual`.

Reply:
You're right. It wasn't apparent because it to enter visual mode, you have to first enter `vicmd` (which works and sets the cursor to a block; but it should have been blinking).

Do you know if there's a reliable way to detect the local modes? Should `vi-cmd-mode `, `visual-mode`, `visual-line-mode` also be wrapped?